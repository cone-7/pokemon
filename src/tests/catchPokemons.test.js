import React from 'react';
import CatchComponent from '../components/catchComponent.js';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

global.fetch = function(){
    return Promise.resolve({json: () => {
            return {
                name: 'ditto',
                sprites: {front_default:'img'
            }
        }}
    });
  }

test('Should render catchPokemons', () => {

  const component = renderer.create(
    <CatchComponent add={() => true}></CatchComponent>,
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

configure({ adapter: new Adapter() });
test('Should click catch', () => {

    const mockCallBack = jest.fn();
    
    const component = shallow(<CatchComponent add={mockCallBack}></CatchComponent>);
    component.setState({pokemon: 'ditto'});
    component.find('#but').simulate('click');
    
    expect(mockCallBack.mock.calls);
  });
