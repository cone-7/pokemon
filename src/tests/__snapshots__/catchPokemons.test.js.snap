// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render catchPokemons 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    id="but"
    onClick={[Function]}
  >
     Catch!!
  </button>
</div>
`;
